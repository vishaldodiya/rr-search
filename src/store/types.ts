import {PinsState} from "./pins/types";
import {DataState} from "./data/types";
import {AnalyticsState} from "./analytics/types";

export interface StoreState {
    pins: PinsState,
    data: DataState,
    analytics: AnalyticsState
};