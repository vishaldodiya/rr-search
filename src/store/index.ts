import {createStore, combineReducers} from "redux";
import * as constants from "../constants";
import DataReducer from "./data/reducer";
import PinsReducer from "./pins/reducer";
import AnalyticsReducer from "./analytics/reducer";
import * as calendar from "../acme-search/calendar.json";
import * as contacts from "../acme-search/contacts.json";
import * as dropbox from "../acme-search/dropbox.json";
import * as slack from "../acme-search/slack.json";
import * as tweet from "../acme-search/tweet.json";

const data = {
    calendar: calendar.calendar,
    contacts: contacts.contacts,
    dropbox: dropbox.dropbox,
    slack: slack.slack,
    tweet: tweet.tweet
};

const reducer = combineReducers({
    pins: PinsReducer,
    data: DataReducer,
    analytics: AnalyticsReducer
});

const store = createStore(reducer);

store.dispatch({
    type: constants.LOAD_DATA,
    list: data
})

store.subscribe(() => console.log(store.getState()));

export default store;