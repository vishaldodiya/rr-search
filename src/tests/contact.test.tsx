import * as React from "react";
import Contact from "../containers/contact";
import {Provider} from "react-redux";
import store from "../store/index";
import * as renderer from 'react-test-renderer';

 
describe('Contact', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Contact data={
                {
                "id": "12345",
                "name": "John Doe",
                "company": "Acme Inc",
                "emails": [
                    "john@acme.co",
                    "doe@gmail.com"
                ],
                "phones": [
                    "650-555-5555",
                    "+44 171 5555 5555"
                ],
                "last_contact": "2019-02-26",
                "uid": "contact-1234"
                }}
            ></Contact>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});