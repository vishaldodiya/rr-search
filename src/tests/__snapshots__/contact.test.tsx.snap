// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact snapshot renders 1`] = `
<article
  className="contact results__card card"
  id="contact-1234"
>
  <div
    className="card__actions"
  >
    <span>
      <i
        className="fas fa-id-card"
      />
    </span>
    <button
      className="card__pin"
      onClick={[Function]}
    >
      <i
        className="fas fa-thumbtack c--green"
      />
    </button>
  </div>
  <h4>
    John Doe
  </h4>
  <div
    className="contact__info"
  >
    <div
      className="contact__info--company contact--item"
    >
      <i
        className="fas fa-briefcase c--gray"
      />
      <span
        className="card__list"
      >
        Acme Inc
      </span>
    </div>
    <div
      className="contact__info--email contact--item"
    >
      <i
        className="fas fa-envelope c--gray"
      />
      <span
        className="card__list"
      >
        <a
          href="mailto:john@acme.co"
        >
          john@acme.co
        </a>
      </span>
      <span
        className="card__list"
      >
        <a
          href="mailto:doe@gmail.com"
        >
          doe@gmail.com
        </a>
      </span>
    </div>
    <div
      className="contact__info--phone contact--item"
    >
      <i
        className="fas fa-phone c--gray"
      />
      <span
        className="card__list"
      >
        <a
          href="tel:650-555-5555"
        >
          650-555-5555
        </a>
      </span>
      <span
        className="card__list"
      >
        <a
          href="tel:+44 171 5555 5555"
        >
          +44 171 5555 5555
        </a>
      </span>
    </div>
    <div
      className="contact__info--last"
    >
      <i
        className="fas fa-phone-volume c--gray"
      />
      <span
        className="card__list"
      >
        February 25, 2019
      </span>
    </div>
  </div>
</article>
`;
